services_registry/endpoints/services_handler.py

```diff
@@ -97,9 +95,13 @@ async def forward_and_process_response(request, qparams_db, path, requested_serv
     return await json_stream(request, response_converted)
 
 
+
 async def response_from_services(path, requested_service_id=None, method='GET', post_data=None):
     LOG.info('-------- response_from_services %s', path)
 
-    return await collect_responses(path, services_list={
-        requested_service_id: SERVICES[requested_service_id]
-    } if requested_service_id is not None else SERVICES)
+    _services = {
+        requested_service_id: conf.services[requested_service_id]
+    } if requested_service_id is not None else conf.services
+
+    collector = Collector(_services)
```

Comment:
This creates a collector for each function call.
If you want, Sabela, we can create _one_ collector, and add a `restrict` function, in a context manager, such that the collector will use the restricted list, and reset back to the full list on exit.
But I think it's the same as creating another Collector, with the restricted list (as above), and let the garbage collector do its thing. Maybe we could add `del collector` in the function, to trigger the GC.